homeassistant/components/tado/sensor.py

```diff
@@ -290,13 +301,16 @@ def _async_update_callback(self):
     @callback
     def _async_update_device_data(self):
         """Handle update callbacks."""
-        try:
-            data = self._tado.data["device"][self.device_id]
-        except KeyError:
-            return
-
-        if self.device_variable == "tado bridge status":
-            self._state = data.get("connectionState", {}).get("value", False)
+        for device in self._tado.devices:
+            if device["serialNo"] == self.device_id:
+                self._device_info = device
+
+        if self.device_variable == "battery state":
+            self._state = self._device_info["batteryState"]
```

Comment:
What kind of battery measurement is this? Normally we should report it in percent and use entity property device class battery.

Reply:
Unfortunately, the Tado API just returns "NORMAL" if the battery is OK, or "LOW" if the batteries are close to depletion.

Reply:
That's a binary state so should be a binary sensor.

Reply:
Ok, I will change that too :)